include/rtweekend.hpp

```diff
@@ -82,14 +82,19 @@ class LocalPseudoRNG {
 
   // Return a random vector in the unit disk of usual norm in plane x, y
   inline vec in_unit_disk() {
-    auto x = float_t(-1., 1.);
+    auto x = float_t(-1.f, 1.f);
     auto maxy = sycl::sqrt(1 - x * x);
     auto y = float_t(-maxy, maxy);
-    return { x, y, 0. };
+    return { x, y, 0.f};
   }
 
  private:
-  trisycl::vendor::trisycl::random::xorshift<> generator;
+  xorshift<> generator;
+};
+
+struct task_context {
+  LocalPseudoRNG rng;
+  sycl::global_ptr<uint8_t> texture_data;
```

Comment:
More generally, I prefer having all the private members first, otherwise there is too much mind state to carry on.
I plan to update globally the project along this, but since you are adding new private members here...

Reply:
It works at least at sw_emu lvl.